api/api.go

```diff
@@ -37,6 +39,43 @@ type SubmissionInfo struct {
 	State string `json:"state"`
 }
 
+// StatusInfo contains information about a user's status on a particular language track.
+type StatusInfo struct {
+	Language string `json:"track_id"`
+	Recent   struct {
+		Problem   string `json:"problem"`
```

Comment:
@kytrinyx @Tonkpils My reasoning behind not making this a separate type now was that I didn't see any current use for that type outside of this, and by removing it I though it would be both premature optimization and potentially confusing. However, the performance benefit of using the pointer instead of the value seems like it might be worth the change. I'll await further input before pulling that out, though.